project/src/test/java/com/google/sps/UserProfileEventsTest.java

```diff
@@ -72,8 +73,44 @@ public void tearDown() {
   }
 
   @Test
-  public void getUserEventsHosting() throws IOException {
-    // TO DO: Add tests for events hosting.
+  public void getUserEventsHosting_zeroEvents() throws IOException {
+    User host = TestUtils.newUser();
+  
+    SpannerTasks.insertOrUpdateUser(host);
+
+    authenticationHelper
+        .setEnvIsLoggedIn(true)
+        .setEnvEmail(EMAIL)
+        .setEnvAuthDomain(AUTH_DOMAIN);
+    Mockito.when(request.getParameter(PARAMETER_EVENT_TYPE)).thenReturn(HOSTING);
+
+    profileEventsServlet.doGet(request, response);
+
+    Assert.assertEquals(
+        CommonUtils.convertToJson(new HashSet<>()).trim(),
+        stringWriter.toString().trim());
+  }
+
+  @Test
+  public void getUserEventsHosting_nonzeroEvents() throws IOException {
+    User host = TestUtils.newUserWithEmail(EMAIL);
+    Event event = TestUtils.newEventWithHost(host);
+    host = host.toBuilder().addEventHosting(event).build();
```

Comment:
Little bit confusing because you have 2 host variables here.

Reply:
It's just an edit of host to attribute the new event to it

Reply:
you can TestUtils.newUserWithEmail(EMAIL).toBuilder().addEventHosting(event).build() which may offer more clarity to person reading code

Reply:
Fair, done